.github/workflows/check-metadata.yml

```diff
@@ -0,0 +1,55 @@
+name: Check Metadata
+'on':
+  pull_request:
+    paths:
+      - '/ct/*.sh'
+      - '/install/*.sh'
+jobs:
+  check-metadata:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
+      - name: Check Metadata Lines in Scripts
+        shell: bash
```

Comment:
I would suggest splitting this up into 5 steps each with their own output (using `echo "errors=$ERROR_COUNT" >> $GITHUB_OUTPUT`: 
 - Find files to check
 - Check Copyright line
 - Check author line(s)
 - Check license line
 - Check source line

And then a 6th step that creates or updates a comment with the state of this workflow depending on the output of the previous steps along the lines of:
```
     - uses: actions/github-script@v7
       if: ${{ steps.authors.outputs.errors != '0' && ... etc }}
        with:
          script: |
            github.rest.issues.createComment({
              issue_number: context.issue.number,
              owner: context.repo.owner,
              repo: context.repo.repo,
              body: 'Your PR currently contain files with headers that do not match our requirements: ${filenames}'
            })
```

But I can understand that that is not necessary for a first version.